### ISSUE TITLE
Image updater infinite loop

### ISSUE THREAD

**Issue by WitoDelnat** [argoproj-labs/argocd-image-updater#736]
**Describe the bug**

- Our CI publishes a releases a `1.11.0` and `1.11` alias. 
- Our Argo is configured to update the image with a git-writeback but we do not auto-sync as this is production. That is, out-of-sync indicates that we can release by pressing sync manually.
- 💥 Argo Image Updater continuously commits alternating between `1.11.0` and `1.11`. You can see an excerpt from the logs below. The result is that every Argo updater run (every ~2 minutes) a commit was created in our repo.
 
```
2024-06-06 18:35:15.496 time="2024-06-06T16:35:15Z" level=info msg="Committing 1 parameter update(s) for application prod-redacted-ui" application=prod-redacted-ui 
2024-06-06 18:35:15.496 time="2024-06-06T16:35:15Z" level=info msg="Successfully updated image 'docker.io/redacted-ui:1.10.6' to 'docker.io/redacted-ui:1.11', but pending spec update (dry run=false)" alias=svc application=prod-redacted-ui image_name=redacted-ui image_tag=1.10.6 registry=docker.io
2024-06-06 18:35:15.496 time="2024-06-06T16:35:15Z" level=info msg="Setting new image to docker.io/redacted-ui:1.11" alias=svc application=prod-redacted-ui image_name=redacted-ui image_tag=1.10.6 registry=docker.io
…
2024-06-06 18:33:10.984 time="2024-06-06T16:33:10Z" level=info msg="Committing 1 parameter update(s) for application prod-redacted-ui" application=prod-redacted-ui
2024-06-06 18:33:10.984 time="2024-06-06T16:33:10Z" level=info msg="Successfully updated image 'docker.io/redacted-ui:1.10.6' to 'docker.io/redacted-ui:1.11.0', but pending spec update (dry run=false)" alias=svc application=prod-redacted-ui image_name=redacted-ui image_tag=1.10.6 registry=docker.io
2024-06-06 18:33:10.984 time="2024-06-06T16:33:10Z" level=info msg="Starting configmap/secret informers"
2024-06-06 18:33:10.983 time="2024-06-06T16:33:10Z" level=info msg="Setting new image to docker.io/redacted-ui:1.11.0" alias=svc application=prod-redacted-ui image_name=redacted-ui image_tag=1.10.6 registry=docker.io
```

**To Reproduce**

- Create a Docker image with version 1.11.0 and 1.11.
- Argo Image updater is running on v0.12.2 and is configured with `image-list: svc=docker.io/redacted-ui:x.x.x-0`, `svc.update-strategy: semver` and writeback git to main.
- Argo auto-sync is disabled

**Expected behavior**

- I prefers the most specific semver (i.e. 1.11.0) or at the minimum choses a stable one.

**Additional context**

- We resolved the bug by removing the `1.11` alias tags from DockerHub and to no longer publish these.

See above

**Version**
Please tell us about the version you encountered the issue with

- v0.12.2

**Logs**
Please paste any relevant logs here

See above

**Comment by clambin** [comment]
seen the same: redis recently released 7.4, but published both 7.4 and 7.4.0 tags.  Image updater is now constantly cycling between both versions.

**Comment by chengfang** [comment]
This seems to be the same issue as https://github.com/argoproj-labs/argocd-image-updater/issues/375, fixed by PR https://github.com/argoproj-labs/argocd-image-updater/pull/559, included in release 0.13.x or later.

When testing with the current branch, I can see the consistent result, i.e., `1.1.0` always returned as the latest from [`1.1.0`, `1.1`]:
```bash
 dist/argocd-image-updater test --registries-conf-path="" --loglevel trace --update-strategy semver --semver-constraint ">=1" quay.io/cfang/argocd-operator

TRAC[0000] Finding out whether to consider 1.1 for being updateable  image=quay.io/cfang/argocd-operator
TRAC[0000] Finding out whether to consider 1.1.0 for being updateable  image=quay.io/cfang/argocd-operator
DEBU[0000] found 2 from 2 tags eligible for consideration  image=quay.io/cfang/argocd-operator
INFO[0000] latest image according to constraint is quay.io/cfang/argocd-operator:1.1.0  application=test image_alias= image_name=quay.io/cfang/argocd-operator registry_url=quay.io
```
Whereas with 0.12.2, I do see both `1.1.0` and `1.1` returned as result across different runs.
